test/e2e/dynamic_provisioning_test.go

```diff
@@ -891,7 +895,7 @@ func (t *dynamicProvisioningTestSuite) defineTests(isMultiZone bool, schedulerNa
 			expectedString = "hello world\r\n"
 		}
 
-		storageClassParameters := map[string]string{"skuName": "StandardSSD_LRS"}
+		storageClassParameters := map[string]string{"skuName": "StandardSSD_LRS", "maxShares": "2"}
```

Comment:
I thought we already had another test with replicas (perhaps its with the scheduler extender tests)? We should use that test case instead of changing this one.

Reply:
Ah, so the test case I was thinking of is https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/7e3ae4379cd7a7b8c2bcbca1ed09554098d735e6/test/e2e/azdiskschedulerextender_test.go#L203. However, this one fails over the pod several times during the test. We can either check between each iteration that the replicas have been created, or just do it at the end. This is a good semi-stress test case, but I wonder if we should add a simple test case to verify replica counts.

I believe that we should do the following:
1. Instead of modifying this existing test case, add a new test case that is essentially a duplicate of this one that sets `maxShares` to 2 and check for replicas, e.g. "Should test pod failover with replicas"
2. Update the scheduler extender test case to verify the expected number of replica attachments after the pod failovers have been performed.